Import onboarding: Update credentials form

bogiii opened the pull request:
<!--
Link a related issue to this PR. If the PR does not immediately resolve the issue,
for example, it requires a separate deployment to production, avoid
using the "fixes" keyword and instead attach the [Status] Fix Inbound label to
the linked issue.
-->

Closes https://github.com/Automattic/wp-calypso/issues/86940

## Proposed Changes

* Adjusted credentials step to match the new design proposal

<img width="432" alt="Screenshot 2024-01-30 at 15 18 07" src="https://github.com/Automattic/wp-calypso/assets/1241413/304361b5-ff94-43a7-9fce-0b37d5197b5b">


## Testing Instructions

<!--
Add as many details as possible to help others reproduce the issue and test the fix.
"Before / After" screenshots can also be very helpful when the change is visual.
-->

* Go to `setup/import-focused?siteSlug={SITE_SLUG}`
* Pass through the flow and reach the Credentials step
* Check if everything looks as proposed in the figma design: RM0R6AwIQRiSVCLdK12pyr-fi-800_14498
* Check if the form works without any problem

<img width="992" alt="Screenshot 2024-01-30 at 15 15 06" src="https://github.com/Automattic/wp-calypso/assets/1241413/2f5fec54-4a46-4526-b4b1-6fb2224bc045">

## Pre-merge Checklist

<!--
Complete applicable items on this checklist **before** merging into trunk. Inapplicable items can be left unchecked.

Both the PR author and reviewer are responsible for ensuring the checklist is completed.
-->

- [ ] Has the general commit checklist been followed? (PCYsg-hS-p2)
- [ ] https://wpcalypso.wordpress.com/devdocs/docs/testing/index.md for your changes?
- [ ] Have you tested the feature in Simple (P9HQHe-k8-p2), Atomic (P9HQHe-jW-p2), and self-hosted Jetpack sites (PCYsg-g6b-p2)?
- [ ] Have you checked for TypeScript, React or other console errors?
- [ ] Have you used memoizing on expensive computations? More info in [Memoizing with create-selector](https://github.com/Automattic/wp-calypso/blob/trunk/packages/state-utils/src/create-selector/README.md) and [Using memoizing selectors](https://react-redux.js.org/api/hooks#using-memoizing-selectors) and [Our Approach to Data](https://github.com/Automattic/wp-calypso/blob/trunk/docs/our-approach-to-data.md)
- [ ] Have we added the "[Status] String Freeze" label as soon as any new strings were ready for translation (p4TIVU-5Jq-p2)?
- [ ] For changes affecting Jetpack: Have we added the "[Status] Needs Privacy Updates" label if this pull request changes what data or activity we track or use (p4TIVU-ajp-p2)?

bogiii committed:
SSH step: add subtitle

bogiii committed:
Credentials form: Hide protocol control if there is only one option

bogiii committed:
Help widget: update position

bogiii committed:
Credentials form: adjust CTA button styles

bogiii committed:
Introduce formModeSwitcher prop; adjust style to match the design

bogiii committed:
Fix unit tests

ouikhuan commented on client/blocks/importer/wordpress/import-everything/pre-migration/credentials.tsx:
```diff
@@ -38,6 +38,17 @@ export function Credentials( props: Props ) {
 				<Title>{ translate( 'You are ready to migrate' ) }</Title>
+				<SubTitle className="onboarding-subtitle--full-width">
+					{ translate(
+						'Provide your SSH server credentials to migrate %(sourceSite)s to %(targetSite)s',
```
Better to add some comments for the translator to explain what the args is here.

bogiii replied:
Agree; added!

ouikhuan reviewed (commented):
Comparing to the design, it looks like the alignment of form and the CTA is not in the center of the page.

![Screen Shot 2024-01-31 at 2 51 06 PM](https://github.com/Automattic/wp-calypso/assets/4074459/6617cf97-e921-486f-8f11-82b68eede724)


And on small screens the width of the input field is not unified, please see screens
hots below:
![Screen Shot 2024-01-31 at 3 10 57 PM](https://github.com/Automattic/wp-calypso/assets/4074459/ab0055fa-eb62-4801-bcc4-9a7cf50900a7)

bogiii committed:
Add additional hint to the translators

bogiii committed:
Fix style issues for mobile devices

bogiii commented:
Hey @ouikhuan,

I've fixed the style issue for the mobile devices and provided info for the translators.
We already discussed centering the screen in the PRs when we introduced this screen, so we agreed to keep the layout position as it is. (to be more precise, it used to be centered, and we decided to go with this version)

Thanks for the review!!!

ouikhuan reviewed (approved):
Thanks for the explanation, looks good to me!

> We already discussed centering the screen in the PRs when we introduced this screen, so we agreed to keep the layout position as it is. (to be more precise, it used to be centered, and we decided to go with this version)

Yeah, I understood, the reason I pointed out is because previously we have two buttons side by side so it looks more balance, but with only one button section now, it looks a little off, but not a blocker here, thanks for handling the feedback!

a8ci18n commented:
This Pull Request is now available for translation here: https://translate.wordpress.com/deliverables/11133224

Some locales (Hebrew, Japanese) have been temporarily machine-translated due to translator availability. All other translations are usually ready within a few days. Untranslated and machine-translated strings will be sent for translation next Monday and are expected to be completed by the following Friday.

Thank you @bogiii for including a screenshot in the description! This is really helpful for our translators.

a8ci18n commented:
Translation for this Pull Request [has now been finished](https://translate.wordpress.com/deliverables/overview/11133224).
